src/IO/S3/S3Capabilities.cpp

```diff
@@ -0,0 +1,34 @@
+#include <IO/S3/S3Capabilities.h>
+
+namespace DB
+{
+
+S3Capabilities::S3Capabilities(const S3Capabilities & src)
+    : S3Capabilities(src.is_batch_delete_supported(), src.support_proxy)
+{
+}
+
+std::optional<bool> S3Capabilities::is_batch_delete_supported() const
+{
+    std::lock_guard lock{mutex};
+    return support_batch_delete;
+}
+
+void S3Capabilities::set_is_batch_delete_supported(std::optional<bool> support_batch_delete_)
```

Comment:
If I am right you never use this function with std::nullopt, that's why you can replace optional<bool> to bool here